File: 496-NextGreaterElementI/496-NextGreaterElementI.py
# Last updated: 07/04/2026, 11:38:52



class Solution:
    def nextGreaterElement(self, nums1, nums2) :
        hash = {}
        for i in range(len(nums2)):
            k = -1
            for j in range(i+1,len(nums2)):
                if nums2[i]<nums2[j] :
                    k= nums2[j]
                    break
                    
            hash[nums2[i]] = k 

        ans = []
        for i in nums1:
            ans.append(hash[i])
        
        return ans
        

File: 496-NextGreaterElementI/496-NextGreaterElementI.cpp
// Last updated: 07/04/2026, 11:38:46
class Solution {
public:
    vector<int> nextGreaterElement(vector<int>& nums1, vector<int>& nums2) {
        unordered_map<int, int> list;
        stack<int> stck;
        vector<int>ans ;

        for(int &i : nums2){
            while (!stck.empty() && stck.top()<i){
                list[stck.top()] = i;
                stck.pop();
            }
            stck.push(i);
        }
        while(!stck.empty()){
           list[stck.top()] = -1;
           stck.pop(); 
        }
        for(int &i: nums1){
           ans.push_back(list[i]); 
        }
        return ans;
    }

};